ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PointSchedule.hs

```diff
@@ -261,6 +271,22 @@ pointSchedulePeers :: PointSchedule -> NonEmpty PeerId
 pointSchedulePeers PointSchedule{ticks = Tick {peers} :| _} =
   getPeerIds peers
 
+-- | Convert 'Peers' to a list of 'Peer'.
+peersList :: Peers a -> NonEmpty (Peer a)
+peersList Peers {honest, others} =
+  honest :| Map.elems others
+
+-- | Construct 'Peers' from values, adding adversary names based on the default schema.
+-- A single adversary gets the ID @adversary@, multiple get enumerated as @adversary N@.
+mkPeers :: a -> [a] -> Peers a
+mkPeers h as =
+  Peers (Peer HonestPeer h) (Map.fromList (mkPeer <$> advs as))
+  where
+    mkPeer (pid, a) = (pid, Peer pid a)
+    advs [a] = [("adversary", a)]
+    advs _   = zip enumAdvs as
+    enumAdvs = (\ n -> PeerId ("adversary " ++ show n)) <$> [1 :: Int ..]
```

Comment:
Also nice!

Reply:
yeah I'm not so happy about the whole `PeerId` thing, that invariant for formatting the adversary name is depended upon in multiple locations

Reply:
Maybe we should just make it an `Int`?

Reply:
that would be better.
it still provides no means of safely referencing the "right" peer with the frequencies, since peers are generated randomly.
don't think we can build a good general framework with this interface.
probably should be a function `Peers TestFrag -> Peers Int` to determine the frequencies, so we don't have to validate afterwards

Reply:
but also we shouldn't sink much time into this, since the slow-honest test variant isn't that useful for us right now